health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/utils/setupCampaignHelpers.js

```
@@ -404,18 +404,34 @@ export const cycleDataRemap=(data)=> {
-    const highestStep = relatedSteps.reduce((max, step) => Math.max(max, parseInt(step.stepCount)), 0);
-    if (isDraft == "true") {
-      const filteredStep = draftFilterStep(totalFormData,campaignConfig);
-      setActive(filteredStep);
-    } else {
+    if (isMicroplanScreen && urlKey) {
+      // Find the step count for the specific key from the URL
+      const matchedStep = campaignConfig?.[0]?.form.find(
+        (step) => step.key === urlKey
+      );
+  
+      highestStep = matchedStep ? parseInt(matchedStep.stepCount) : 0;
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)